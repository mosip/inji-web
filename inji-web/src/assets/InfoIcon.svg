<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M8 1.76779C9.13858 1.76779 10.1773 2.04744 11.1161 2.60674C12.0549 3.14607 12.814 3.90512 13.3933 4.88389C13.9526 5.86267 14.2322 6.90137 14.2322 8C14.2322 9.09862 13.9526 10.1373 13.3933 11.1161C12.814 12.0949 12.0549 12.8539 11.1161 13.3933C10.1773 13.9526 9.13858 14.2322 8 14.2322C6.86142 14.2322 5.82272 13.9526 4.8839 13.3933C3.94507 12.8539 3.18602 12.0949 2.60674 11.1161C2.04744 10.1373 1.76779 9.09862 1.76779 8C1.76779 6.90137 2.04744 5.86267 2.60674 4.88389C3.18602 3.90512 3.94507 3.14607 4.8839 2.60674C5.82272 2.04744 6.86142 1.76779 8 1.76779ZM8 15.6704C9.39825 15.6504 10.6767 15.3109 11.8352 14.6517C12.9938 13.9725 13.9326 13.0337 14.6517 11.8352C15.3308 10.6167 15.6704 9.33833 15.6704 8C15.6704 6.66167 15.3308 5.38327 14.6517 4.16479C13.9326 2.96629 12.9938 2.02746 11.8352 1.34831C10.6767 0.689137 9.39825 0.349561 8 0.329586C6.60175 0.349561 5.32335 0.689137 4.1648 1.34831C3.00624 2.02746 2.06742 2.96629 1.34832 4.16479C0.669165 5.38327 0.32959 6.66167 0.32959 8C0.32959 9.33833 0.669165 10.6167 1.34832 11.8352C2.06742 13.0337 3.00624 13.9725 4.1648 14.6517C5.32335 15.3109 6.60175 15.6504 8 15.6704ZM6.8015 10.397H6.0824V11.8352H6.8015H9.9176V10.397H9.1985H8.9588V7.7603V7.0412H8.2397H6.0824V8.4794H6.8015H7.5206V10.397H6.8015ZM8.9588 6.0824V4.16479H7.0412V6.0824H8.9588Z" fill="black"/>
</svg>
