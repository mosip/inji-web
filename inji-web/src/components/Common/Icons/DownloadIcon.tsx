import React from "react";

export function DownloadIcon({ gradient = false, style, ...props }: { gradient?: boolean, style?: React.CSSProperties, testId: string} & React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      style={style}
      aria-label={"icon-download"}
      data-testid={props.testId}
      {...props}
    >
      {gradient && (
        <defs>
          <linearGradient id="downloadGradient" x1="0" y1="0" x2="24" y2="24" gradientUnits="userSpaceOnUse">
            <stop stopColor="#6a11cb"/>
            <stop offset="1" stopColor="#2575fc"/>
          </linearGradient>
        </defs>
      )}
      <path
        d="M12 15.7885L7.73075 11.5192L8.78475 10.4348L11.25 12.9V4.5H12.75V12.9L15.2153 10.4348L16.2693 11.5192L12 15.7885ZM6.30775 19.5C5.80258 19.5 5.375 19.325 5.025 18.975C4.675 18.625 4.5 18.1974 4.5 17.6923V14.9808H6V17.6923C6 17.7692 6.03208 17.8398 6.09625 17.9038C6.16025 17.9679 6.23075 18 6.30775 18H17.6923C17.7692 18 17.8398 17.9679 17.9038 17.9038C17.9679 17.8398 18 17.7692 18 17.6923V14.9808H19.5V17.6923C19.5 18.1974 19.325 18.625 18.975 18.975C18.625 19.325 18.1974 19.5 17.6923 19.5H6.30775Z"
        fill={gradient ? "url(#downloadGradient)" : "currentColor"}
      />
    </svg>
  );
}