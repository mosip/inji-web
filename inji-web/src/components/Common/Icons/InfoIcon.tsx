import React from 'react';
import InfoIconSrc from '../../../assets/InfoIcon.svg';

interface InfoIconProps {
  className?: string;
  testId?: string;
  alt?: string;
}

export const InfoIcon: React.FC<InfoIconProps> = ({
  className = '',
  testId = 'icon-info',
  alt = 'Info'
}) => {
  return (
    <img
      data-testid={testId}
      src={InfoIconSrc}
      alt={alt}
      className={className}
    />
  );
};