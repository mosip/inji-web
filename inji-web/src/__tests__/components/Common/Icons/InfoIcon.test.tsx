import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {InfoIcon} from '../../../../components/Common/Icons/InfoIcon';

jest.mock('../../../../assets/InfoIcon.svg', () => 'mocked-info-icon-path');

describe('InfoIcon Component', () => {
    const defaultProps = {
        testId: 'icon-info',
        alt: 'Info'
    };

    const renderInfoIcon = (props = {}) => {
        return render(<InfoIcon {...defaultProps} {...props} />);
    };

    it('matches snapshot with default props', () => {
        const {asFragment} = renderInfoIcon();

        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with custom props', () => {
        const {asFragment} = renderInfoIcon({
            className: 'custom-class mt-2',
            testId: 'custom-info-icon',
            alt: 'Custom Info Text'
        });

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders with default test ID', () => {
        renderInfoIcon();

        expect(screen.getByTestId(defaultProps.testId)).toBeInTheDocument();
    });

    it('renders with custom test ID', () => {
        const customTestId = 'my-info-icon-id';

        renderInfoIcon({testId: customTestId});

        expect(screen.getByTestId(customTestId)).toBeInTheDocument();
    });

    it('renders with custom className', () => {
        const customClass = 'test-custom-class';
        renderInfoIcon({className: customClass});
        
        const icon = screen.getByTestId(defaultProps.testId);
        
        expect(icon).toHaveClass(customClass);
    });

    it('renders without any class when no className is provided', () => {
        renderInfoIcon({className: ''});
        
        const icon = screen.getByTestId(defaultProps.testId);
        
        expect(icon.className).toBe('');
    });

    it('sets correct default alt text', () => {
        renderInfoIcon();
        
        const icon = screen.getByTestId(defaultProps.testId);
       
        expect(icon).toHaveAttribute('alt', defaultProps.alt);
    });

    it('sets custom alt text when provided', () => {
        const customAltText = 'Custom Info Text';
        renderInfoIcon({alt: customAltText});

        const icon = screen.getByTestId(defaultProps.testId);
        
        expect(icon).toHaveAttribute('alt', customAltText);
    });

    it('uses correct image source', () => {
        renderInfoIcon();

        const icon = screen.getByTestId(defaultProps.testId);

        expect(icon).toHaveAttribute('src', 'mocked-info-icon-path');
    });
});
