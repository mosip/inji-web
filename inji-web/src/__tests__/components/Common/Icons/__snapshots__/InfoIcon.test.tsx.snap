// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoIcon Component matches snapshot with custom props 1`] = `
<DocumentFragment>
  <img
    alt="Custom Info Text"
    class="custom-class mt-2"
    data-testid="custom-info-icon"
    src="mocked-info-icon-path"
  />
</DocumentFragment>
`;

exports[`InfoIcon Component matches snapshot with default props 1`] = `
<DocumentFragment>
  <img
    alt="Info"
    class=""
    data-testid="icon-info"
    src="mocked-info-icon-path"
  />
</DocumentFragment>
`;
